js/pages/data-sources/data-sources.html

```diff
@@ -0,0 +1,83 @@
+<div class="wrapperTitle">
+	<span data-bind="if: !loadingReport()">
+		<i class="fa fa-area-chart"></i>
+	</span> Data Source Reports
+</div>
+
+<div data-bind="ifnot: canViewCdmResults">
+	<div class="padded">
+		<access-denied params="isAuthenticated: $component.isAuthenticated"></access-denied>
+	</div>
+</div>
+
+<div data-bind="if: canViewCdmResults">
+	<!-- ko if: showSelectionArea -->
+	<div class="reportControls pad-5">
+		<div class="row">
+			<div class="col-xs-12">
+				<select class="form-control" data-bind="css : { invalid:  currentSource()==undefined }, options: sources, optionsText:'sourceName', value: currentSource, optionsCaption:'Select a Source'"></select>
+			</div>
+		</div>
+		<div class="row">
+			<div class="col-xs-12">
+				<select class="form-control" data-bind="css : { invalid:  currentReport()==undefined }, options: reports, optionsCaption:'Select a Report', optionsText:'name', value: currentReport"></select>
+			</div>
+		</div>
+		<div class="clear"></div>
+	</div>
+	<!-- /ko -->
+
+	<loading params="status: 'loading report'" data-bind="visible: loadingReport() && ! hasError()">
+	</loading>
+
+	<div class="paddedWrapper text-center error" data-bind="visible: hasError()">
+		<div>
+			<i class="fa fa-exclamation-circle"></i>
+		</div>
+		<div>error loading report</div>
+	</div>
+
+	<div id="report" class="flexed" data-bind="if:  currentReport() && ! hasError()">
+		<div class="row pad-10 report">
+			<!-- ko if: $component.currentReport().name == 'Dashboard'-->
```

Comment:
Have you tried the component binding without a container element? 
See: http://knockoutjs.com/documentation/component-binding.html

 Like so:
```
<!-- ko component: {
    name: $component.currentReport().name,
    params: { context: $component }
} -->
<!-- /ko -->
```

The component will reload based on the value of currentReport().name (or you can introduce a new property on the report called 'componentName' in the case that you want to keep the name of the report separate from the name of the component for this report.

Reply:
I never knew there's such binding, thanks for the hint!